### PR TITLE
Keypad

### PR DESCRIPTION
**Pull request by Kavikick**
Created Keypad_TC class, and made some compatibility changes to the existing framework to accommodate the Keypad library itself.

### CODE REVIEW

File: src/TankControllerLib.cpp [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #include "UIState/MainMenu.h"
 #include "UIState/UIState.h"
 
-const char NO_KEY = '\0';  // Move to LiquidCrystal?
+// const char NO_KEY = '\0';  // Move to LiquidCrystal?
```
Comment (prestoncarman) [inline src/TankControllerLib.cpp]: Do not include commented out code in your PR.

File: src/TankControllerLib.h [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 #include "Devices/DateTime_TC.h"
 #include "Devices/EthernetServer_TC.h"
+#include "Devices/Keypad_TC.h"
```
Comment (prestoncarman) [inline src/TankControllerLib.h:6]: Is this needed at this time?

**Review by prestoncarman — CHANGES_REQUESTED** [review]

File: src/TankControllerLib.h (hunk 2 above)
Comment (Kavikick) [reply src/TankControllerLib.h:6]: Upon moving it to TankControllerLib.cpp the tests wouldn't compile anymore.

File: test/Keypad.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  assertNotNull(singleton1);
+  Keypad_TC* singleton2 = nullptr;
+  singleton2 = Keypad_TC::instance();
+  assertNotNull(singleton2);
+  assertEqual(singleton1, singleton2);
+}
+
+unittest(getKey) {
+  Keypad_TC* testKeypad = Keypad_TC::instance();
+  Keypad* puppet = testKeypad->_getPuppet();
```
Comment (jgfoster) [inline test/Keypad.cpp:19]: Could we confirm that if there is nothing available that the puppet returns NO_KEY?

**Review by jgfoster — CHANGES_REQUESTED** [review]
Request one more minor test.

File: test/Keypad.cpp (hunk 3 above)
Comment (Kavikick) [reply test/Keypad.cpp:19]: Testing has been added, but it relies on behavior in the CI version. The tests will pass once the NO_KEY pull request is accepted on that end.

**Comment by jgfoster** [comment]
Any idea why `getKey()` and `waitForKey()` are failing?

**Comment by Kavikick** [comment]
Because the Keypad_CI::getKey() which our wrapper depends on returns a -1 instead of NO_KEY. I made a pull request with the change, and when tested it passes. I was considering changing the test to accept the -1 but I feel like changing the behavior to be more accurate would be best.

**Review by jgfoster — APPROVED** [review]

**Review by prestoncarman — APPROVED** [review]
LGTM

**Pull request merged**